src/rev_ai/generic_api_client.py

```diff
@@ -0,0 +1,167 @@
+from .baseclient import BaseClient
+
+try:
+    from urllib.parse import urljoin
+except ImportError:
+    from urlparse import urljoin
+
+
+class GenericApiClient(BaseClient):
+    """Generic client which handles logic for making requests to almost any Rev AI Api.
+    Intended to be inherited and extended by a specific client per API"""
+
+    def __init__(self, access_token, api_name, api_version, parse_job_info, parse_job_result):
+        """Constructor
+
+        :param access_token: access token which authorizes all requests and links them to your
+                             account. Generated on the settings page of your account dashboard
+                             on Rev AI.
+        :param api_name: name of the api to submit to
+        :param api_version: version of the api to submit to
+        :param parse_job_info: method to be used to parse job information
+        :param parse_job_result: method to be used to parse job results
+        """
+
+        BaseClient.__init__(self, access_token)
+        self.base_url = 'https://api.rev.ai/{0}/{1}/'.format(api_name, api_version)
+        self.parse_job_info = parse_job_info
+        self.parse_job_result = parse_job_result
+
+    def _submit_job(self,
+                    metadata=None,
+                    callback_url=None,
+                    delete_after_seconds=None,
+                    language=None,
+                    **kwargs):
+        """Submit a job to the api. This method is special in that it is intended to be hidden by
+        the implementation this is done because python standard is to pass options individually
+        instead of as an object and our true clients should match this standard
+
+        :param metadata: info to associate with the transcription job
+        :param callback_url: callback url to invoke on job completion as
+                             a webhook
+        :param delete_after_seconds: number of seconds after job completion when job is auto-deleted
+        :param language: specify language using the one of the supported ISO 639-1 (2-letter) or
+            ISO 639-3 (3-letter) language codes as defined in the API Reference
+        :returns: Job info object
+        :raises: HTTPError
+        """
+        payload = {}
+        for key, value in kwargs.items():
+            if value:
```

Comment:
won't this not work if the values are intentionally false-y, like `skip_punctuation=false` 
```suggestion
            if value is not None:
```

Reply:
Good catch. I'll do an explicit None check